File: app/src/main/java/com/shannan/instawordcounter/data/NetworkCall.java
package com.shannan.instawordcounter.data;

import android.util.Log;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.net.HttpURLConnection;
import java.net.URL;

public class NetworkCall {

    public String makeNetworkCall(String urlString) {
        HttpURLConnection connection = null;
        String response = null;
        InputStream is = null;

        try {
            URL url = new URL(urlString);
            connection = (HttpURLConnection) url.openConnection();
            connection.connect();
            is = connection.getInputStream();
            response = readIt(is);
            Log.d("response", response);
        } catch (Exception e) {
            e.printStackTrace();
        } finally {
            try {
                if (is != null) {
                    connection.disconnect();
                    is.close();
                }
            } catch (Exception e) {
                e.printStackTrace();
            }
        }
        return response;
    }

    private String readIt(InputStream iStream) throws Exception {
        String singleLine;
        StringBuilder totalLines = new StringBuilder(iStream.available());
        BufferedReader reader = null;
        try {
            reader = new BufferedReader(new InputStreamReader(iStream));

            while ((singleLine = reader.readLine()) != null) {
                totalLines.append(singleLine);
            }
        } catch (Exception e) {
            e.printStackTrace();
        } finally {
            if (reader != null) {
                try {
                    reader.close();
                } catch (IOException e) {
                    e.printStackTrace();
                }
            }
        }
        return totalLines.toString();
    }

}

File: app/src/main/java/com/shannan/instawordcounter/data/remote/NetworkCall.kt
package com.shannan.instawordcounter.data.remote

import java.io.BufferedReader
import java.io.IOException
import java.io.InputStream
import java.io.InputStreamReader
import java.net.HttpURLConnection
import java.net.URL

class NetworkCall {
    fun makeNetworkCall(urlString: String?): String? {
        var connection: HttpURLConnection? = null
        var response: String? = null
        var inputStream: InputStream? = null
        try {
            val url = URL(urlString)
            connection = url.openConnection() as HttpURLConnection
            connection.connect()
            inputStream = connection.inputStream
            response = readIt(inputStream)
        } catch (e: Exception) {
            e.printStackTrace()
        } finally {
            try {
                if (inputStream != null) {
                    connection!!.disconnect()
                    inputStream.close()
                }
            } catch (e: Exception) {
                e.printStackTrace()
            }
        }
        return response
    }

    @Throws(Exception::class)
    private fun readIt(iStream: InputStream?): String {
        var singleLine: String?
        val totalLines = StringBuilder(iStream!!.available())
        var reader: BufferedReader? = null
        try {
            reader = BufferedReader(InputStreamReader(iStream))
            while (reader.readLine().also { singleLine = it } != null) {
                totalLines.append(singleLine)
            }
        } catch (e: Exception) {
            e.printStackTrace()
        } finally {
            if (reader != null) {
                try {
                    reader.close()
                } catch (e: IOException) {
                    e.printStackTrace()
                }
            }
        }
        return totalLines.toString()
    }
}